grammar/haxe.bnf

```diff
@@ -436,7 +436,7 @@ private namedTypeOrAnonymous ::= [optionalMark] [componentName ':'] typeOrAnonym
 private namedFunctionType ::= [optionalMark] [componentName ':'] functionType
 private optionalFunctionTypeWithoutName ::= optionalMark '(' functionType ')' {pin=3}
 
-private oldFunctionType ::= oldFunctionTypeArgumentsList functionReturnType {pin=2}
+private oldFunctionType ::= oldFunctionTypeArgumentsList (functionReturnType |'(' functionReturnType ')') {pin=2}
```

Comment:
The ordering of parens-first match vs. parens-last is different between line 310 and here.  Do you think there's any advantage of one over the other?  (They are functionally equivalent, given that `functionReturnType ::= typeWrapper`.)

Reply:
i don't remember if there was a reason for that, i changed it and tested it with a few class that used to be problematic  and they all look like they parse just fine, so i guess  the order does not make a difference.

Reply:
Yes, I would make them consistent, if nothing breaks.  I like this order better, just in case the function type needs to have its own parens:  `(functionType | '(' functionType ')' )`  (as you have it here).